squid/entrypoint.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Squid OCI image entrypoint
+
+# This entrypoint aims to forward the squid logs to stdout to assist users of
+# common container related tooling (e.g., kubernetes, docker-compose, etc) to
+# access the service logs.
+
+# Moreover, it invokes the squid binary, leaving all the desired parameters to
+# be provided by the "command" passed to the spawned container. If no command
+# is provided by the user, the default behavior (as per the CMD statement in
+# the Dockerfile) will be to use default configuration [1] and run
+# squid with the "-NYC" options to mimic the behavior of the Ubuntu provided
+# systemd unit.
+
+# [1] The default configuration is changed in the Dockerfile to allow local
+# network connections. See the Dockerfile for further information.
+
+# re-create snakeoil self-signed certificate removed in the build process
+if [ ! -f /etc/ssl/private/ssl-cert-snakeoil.key ]; then
+    /usr/sbin/make-ssl-cert generate-default-snakeoil --force-overwrite > /dev/null 2>&1
+fi
+
```

Comment:
Re: previous comment, if turns out we can remove the default cert step, then all that would be in this file is printing the logs to stdout and launching squid.

If that was the case i'd delete everything else, just have those lines left, and add a note about squid not logging anything by default to stdout, and it being a launch script.

Dependent on previous comment of course :)

Reply:
And also that it requires root to run :)

Reply:
@mritunjaysharma394 - squid does drops uid/gid to `nobody` if it detects it is running as root. There are two settings it uses `cache_effective_user` and `cache_effective_group` in `squid.conf` to configure a different user.

Reply:
Got it @jamonation that means we can still solve the problem of running it as non-root but I am still confused about snakeoil self-signed certificate thing, maybe I should give it a try without it and see

Reply:
@mritunjaysharma394 - yes squid's config that's being shipped here will run by default with no TLS. So If you're adding tests, a separate pipeline step to generate a key/cert and run with a squid.conf that uses them would be a substitute for the Debian/Ubuntu snakeoil cert.